lib/src/group_changes.dart

```diff
@@ -0,0 +1,307 @@
+    document.getElementById(id + "-off").style.display = "block";
+  } else {
+    document.getElementById(id + "-off").style.display = "none";
+    document.getElementById(id + "-on").style.display = "block";
+  }
+}
+
+function showInline(id) {
+  if (document.getElementById(id + "-off").style.display == "none") {
+    document.getElementById(id + "-on").style.display = "none";
+    document.getElementById(id + "-off").style.display = "inline";
+  } else {
+    document.getElementById(id + "-off").style.display = "none";
+    document.getElementById(id + "-on").style.display = "inline";
+  }
+}
+</script>
+</head><body><h1>Results Feed</h1>
+''';
+
+String postlude() => '''</body></html>''';
+
+String htmlPage(Map<int, Map<int, Map<String, List<ByConfigSetData>>>> data,
+    List<String> hashes, List<String> commitData, List<String> reviewLinks) {
+  StringBuffer page = StringBuffer(prelude());
+
+  page.write("<table>");
+  var after = MinMax();
+  data.keys.forEach(after.add);
+  for (var afterKey = 0; afterKey <= after.max; ++afterKey) {
+    // Info from git about this commit:
+    page.write("<span class='commit'><a href='${reviewLinks[afterKey]}'>"
+        "${hashes[afterKey].substring(0, 8)}</a>&nbsp;&nbsp;"
+        "${commitData[afterKey]}</span><br>");
+    if (!data.containsKey(afterKey)) continue;
+    final beforeKeys = data[afterKey].keys.toList()..sort();
+    for (var beforeKey in beforeKeys) {
+      void writeChange(int i) {
+        page.write(
+            "<span class='indent'><a href='${reviewLinks[i]}'>${hashes[i]}</a>"
+            " ${commitData[i]}</span><br>");
+      }
+
+      final size = beforeKey - afterKey;
+      page.write("<div class='blamelist'><strong>Blamelist:</strong><br>");
+      const int summarize_size = 6;
+      if (size < summarize_size) {
+        for (int i = afterKey; i < beforeKey; ++i) {
+          writeChange(i);
+        }
+      } else {
+        var id = '$afterKey-$beforeKey';
+        page.write("<div onclick='showBlamelist(\"$id\")'>");
+        for (int i = afterKey; i < afterKey + 3; ++i) {
+          writeChange(i);
+        }
+        page.write("<div class='expand_off' id='$id-off'>"
+            "<span class='indent'>...</span></div>");
+        page.write("<div class='expand_on' id='$id-on' style='display:none'>");
+        for (int i = afterKey + 3; i < beforeKey - 1; ++i) {
+          writeChange(i);
+        }
+        page.write("</div>");
+        writeChange(beforeKey - 1);
+        page.write("</div>");
+      }
+      page.write("</div>");
+
+      var configSetKeys = data[afterKey][beforeKey].keys.toList()..sort();
+      for (var configSetKey in configSetKeys) {
+        page.write("<table>");
+        final tests = data[afterKey][beforeKey][configSetKey]
+          ..sort((a, b) => a.test.compareTo(b.test));
+        for (final test in tests) {
+          var testclass = "nomatch";
+          if (test.change.endsWith("&#x2714;</strong>")) {
```

Comment:
Consider having a bool matches in the data instead. Structure the data and move the presentation here.